gazebo_ros/src/gazebo_ros_api_plugin.cpp

```diff
@@ -204,12 +203,13 @@ void GazeboRosApiPlugin::loadGazeboRosApiPlugin(std::string world_name)
   pub_link_states_connection_count_ = 0;
   pub_model_states_connection_count_ = 0;
 
+  // Manage clock for simulated ros time
+  pub_clock_ = nh_->advertise<rosgraph_msgs::Clock>("/clock",10);
+
```

Comment:
By moving this above `advertiseServices`, we get exactly the same behavior as before, but without the code duplication.